www/js/diary/infinite_scroll_trip_item.js

```diff
@@ -0,0 +1,159 @@
+                                                        'emission.stats.clientstats',
+                                                        'emission.incident.posttrip.manual'
+])
+
+.directive("infiniteScrollTripItem", function(){
+    return{
+      restrict: 'E',
+      scope: {
+        trip: '='
+      },
+      controller: 'TripItemCtrl',
+      templateUrl: 'templates/diary/trip_list_item.html'
+    };
+  })
+
+  .controller("TripItemCtrl", function($scope, $injector, $ionicPlatform,
+                                        $state, leafletMapEvents, 
+                                        nzTour, Timeline, DiaryHelper, SurveyOptions,
+                                        Config, $ionicScrollDelegate
+                                        ){
+    console.log("Trip Item Controller called");
+    const DEFAULT_ITEM_HT = 274;
+    $scope.surveyOpt = SurveyOptions.MULTILABEL;
+    $scope.itemHt = DEFAULT_ITEM_HT;
+
+    // Added function from infiniteDiaryListCtrl
+    $scope.showDetail = function($event) {
+      $state.go("root.main.inf_scroll-detail", {
+          tripId: $scope.trip.id
+      });
+      console.log("Testing if showDetail has the trip defined: ", $scope.trip);
+    }
+
+    console.log("Trip before tripgj transformation ", $scope.trip);
+
+    // var getTripComponents = function(trip) {
+    //   console.log("getSections("+trip+") called");
+    //   var startPlace = null;
+    //   var endPlace = null;
+    //   var stopList = [];
+    //   var sectionList = [];
+    //   trip.data.features.forEach(function(feature, index, array) {
+    //         // console.log("Considering feature " + JSON.stringify(feature));
+    //         switch (feature.type) {
+    //           case "Feature":
+    //           switch(feature.properties.feature_type) {
+    //             case "start_place":
+    //             startPlace = feature;
+    //             break;
+    //             case "end_place":
+    //             endPlace = feature;
+    //             break;
+    //             case "stop":
+    //             stopList.push(feature);
+    //             break;
+    //           }
+    //           break;
+    //           case "FeatureCollection":
+    //           feature.features.forEach(function (item, index, array) {
+    //             if (angular.isDefined(item.properties) && angular.isDefined(item.properties.feature_type)) {
+    //                     // console.log("Considering feature with type " + item.properties.feature_type);
+    //                     if (item.properties.feature_type == "section") {
+    //                       console.log("FOUND section" + item + ", appending");
+    //                       sectionList.push(item);
+    //                     }
+    //                   }
+    //                 });
+    //         }
+    //       });
+    //   return [startPlace, endPlace, stopList, sectionList];
+    // };
+
```

Comment:
When I was attempting to read in the trip percentages, this was a function that I had pulled from diaryHelper to see if I could populate the necessary attributes.

Reply:
ok but since it didn't work, clean it up for the next implementer. Having tons of commented out code sitting around makes it harder for people to see what we have now and work with it.

Reply:
This was completed

**_Confirmed this change as complete_**